### PR TITLE
dns implementation as per spec 1.1

### PR DESCRIPTION
**Pull request by WAvdBeek**

### CODE REVIEW

File: port/linux/dns-sd.c [hunk 1]
```diff
@@ -56,56 +56,29 @@ knx_publish_service(char *serial_no, uint32_t iid, uint32_t ia, bool pm)
     snprintf(serial_no_subtype, sizeof(serial_no_subtype), serial_format_string,
              serial_no);
 
-    char *pm_subtype = "--subtype=_pm._sub._knx._udp";
+    char *installation_format_string = "--subtype=_ia%x-%x._sub._knx._udp";
+    snprintf(installation_subtype, sizeof(installation_subtype), serial_format_string,
+             ia, iid);
```
Comment (tiniuclx) [inline port/linux/dns-sd.c:61]: This is incorrect - the format string takes two arguments (%x and %x for the IA) but it is also passed the serial format string.

For the Linux API, i think each argument (so each subtype) really needs to be in it's own argument to execlp, which is the reason we had `if` statements below checking whether ia & iid are set.

**Review by tiniuclx — CHANGES_REQUESTED** [review]
Windows implementation looks good! Linux one isn't quite there

File: port/linux/dns-sd.c (hunk 1 above)
Comment (tiniuclx) [reply port/linux/dns-sd.c:61]: Note that the IA subtype shouldn't be present at all. I wonder what happens if we set it to ```""``` when that is the case (like what you did for the pm) - hopfully `execlp` below just doesn't handle that argument, but this specific usage is not well documented

**Review by tiniuclx — APPROVED** [review]
Let's merge it

**Comment by tiniuclx** [comment]
~format

File: port/linux/dns-sd.c [hunk 2]
```diff
@@ -56,56 +56,29 @@ knx_publish_service(char *serial_no, uint32_t iid, uint32_t ia, bool pm)
     snprintf(serial_no_subtype, sizeof(serial_no_subtype), serial_format_string,
              serial_no);
 
-    char *pm_subtype = "--subtype=_pm._sub._knx._udp";
+    char *installation_format_string = "--subtype=_ia%x-%x._sub._knx._udp";
+    snprintf(installation_subtype, sizeof(installation_subtype),
+             ia, iid);
```
[the remark this answers is not in the document]
Comment (WAvdBeek) [reply port/linux/dns-sd.c]: yes, there is no clause that that should not be done

**Pull request merged**